docs/question.md

```diff
@@ -544,3 +538,90 @@ _Note:_ Data stored under the `"submitted_answers"` key in the data dictionary m
 element sometimes stores very large integers as strings instead of the Python `int` type used in most cases. The best practice for custom grader
 code in this case is to always cast the data to the desired type, for example `int(data["submitted_answers"][name])`. See the
 [PrairieLearn elements documentation](elements.md) for more detailed discussion related to specific elements.
+
+## Initializing parameters from outside the question
+
+The `data["params"]` dictionary normally used to pass parameters to `question.html` can be "seeded" from outside the question's files, allowing some portion of its randomization to be set by an assessment, course instance, or course configuration file. Such "`questionParams`" can change the behavior of a question without needing to alter its source code, so a question can be reused on different assessments but with a different configuration of parameters to alter its actual variants as needed.
+
+A `questionParams` is a JSON object that can be placed in three different PrairieLearn file types. When a question's `generate()` function is called, the `data["params"]` dictionary will be pre-filled with the key-value pairs defined by `questionParams` in these locations:
```

Comment:
I think we want to change this behavior to copy it into a sub-dictionary under a special key: https://github.com/PrairieLearn/PrairieLearn/pull/3848#discussion_r947765511